cw-orch-daemon/src/builder.rs

```diff
@@ -167,40 +191,32 @@ impl DaemonAsyncBuilder {
             }
         };
         // if mnemonic provided, use it. Else use env variables to retrieve mnemonic
-        let sender_options = self.sender_options.clone();
 
-        let sender = match self.sender.clone() {
-            Some(sender) => match sender {
-                SenderBuilder::Mnemonic(mnemonic) => Sender::from_mnemonic_with_options(
-                    chain_info.clone(),
-                    GrpcChannel::connect(&chain_info.grpc_urls, &chain_info.chain_id).await?,
-                    &mnemonic,
-                    sender_options,
-                )?,
-                SenderBuilder::Sender(mut sender) => {
-                    sender.set_options(self.sender_options.clone());
-                    sender
-                }
-            },
-            None => Sender::new_with_options(
-                chain_info.clone(),
-                GrpcChannel::connect(&chain_info.grpc_urls, &chain_info.chain_id).await?,
-                sender_options,
-            )?,
+        let sender = if let Some(sender) = &self.sender {
+            let mut sender = sender.clone();
+            sender.set_options(self.sender_options.clone());
```

Comment:
I don't think this will give use the expected behaviour. 

Let's say we want to use a custom sender. First, we're required to build the sender ourselves (there's no way to set the custom sender options ATM). So if we build it and configure it to our liking, then we'll provide it to this builder. The builder will take and and then **apply the `SenderOptions::Default` value to the sender that we just configured. This would be extremely confusing to run into. 

Additionally if we add the ability to set the options for the sender, we can still confuse the user with this behaviour. 

And it its current state the `else` part of this builder will only be called with `Sender` being the default `Wallet`. The `Sender::build` function is only called in this location and adds a lot of complexity to the `Sender` trait.